lib/rails_state_machine/model.rb

```diff
@@ -45,27 +45,5 @@ def state_machine_state_managers
         state_machine_state_manager(state_attribute)
       end
     end
-
-    def prepare_state_event_change(attributes)
-      if ActiveRecord::VERSION::STRING < '5.2' && saved_changes?
```

Comment:
It's a bit odd we didn't cover this with Rails 5.1 tests, but I'm totally fine with dropping support for Rails <5.2 (just remember to adjust dependencies).